Source/WebKit/WebProcess/WebPage/UnifiedTextReplacementController.cpp

```diff
@@ -406,39 +407,50 @@ void UnifiedTextReplacementController::textReplacementSessionDidReceiveEditActio
         return;
     }
 
+    if (m_replacements.isEmpty())
+        return;
+
+    auto contents = liveRange->cloneContents();
+    if (contents.hasException()) {
+        RELEASE_LOG(UnifiedTextReplacement, "UnifiedTextReplacementController::textReplacementSessionDidReceiveEditAction (%s) => exception when cloning contents", uuid.toString().utf8().data());
+        return;
+    }
+
     switch (action) {
-    case WebKit::WebTextReplacementData::EditAction::Undo:
-    case WebKit::WebTextReplacementData::EditAction::Redo: {
-        if (m_replacements.isEmpty())
-            return;
+    case WebKit::WebTextReplacementData::EditAction::Undo: {
+        auto originalFragment = m_originalDocumentNodes.get(uuid);
```

Comment:
I think we need to start using `Ref` rather than `auto` here, per current smart pointer guidelines. Similar instances throughout this change.

Reply:
```
In file included from /Users/richardrobinson/Build/Products/Debug/DerivedSources/WebKit/unified-sources/UnifiedSource118.cpp:2:
/Users/richardrobinson/safari/OpenSource/Source/WebKit/WebProcess/WebPage/UnifiedTextReplacementController.cpp:168:9: error: no viable constructor or deduction guide for deduction of template arguments of 'Ref'
  168 |     Ref liveRange = m_contextRanges.get(uuid);
In file included from /Users/richardrobinson/Build/Products/Debug/DerivedSources/WebKit/unified-sources/UnifiedSource118.cpp:1:
In file included from /Users/richardrobinson/safari/OpenSource/Source/WebKit/WebProcess/WebPage/PageBanner.cpp:27:
In file included from /Users/richardrobinson/safari/OpenSource/Source/WebKit/WebProcess/WebPage/PageBanner.h:28:
In file included from /Users/richardrobinson/safari/OpenSource/Source/WebKit/Shared/API/APIObject.h:29:
In file included from /Users/richardrobinson/Build/Products/Debug/usr/local/include/wtf/RefPtr.h:28:
/Users/richardrobinson/Build/Products/Debug/usr/local/include/wtf/Ref.h:85:5: note: candidate function [with T = WebCore::Range *, _PtrTraits = WTF::RawPtrTraits<WebCore::Range *>, RefDerefTraits = WTF::DefaultRefDerefTraits<WebCore::Range *>] not viable: expects an lvalue for 1st argument
   85 |     Ref(T& object)
/Users/richardrobinson/Build/Products/Debug/usr/local/include/wtf/Ref.h:123:5: note: candidate template ignored: couldn't infer template argument 'T'
  123 |     Ref(HashTableDeletedValueType) : m_ptr(PtrTraits::hashTableDeletedValue()) { }
/Users/richardrobinson/Build/Products/Debug/usr/local/include/wtf/Ref.h:126:5: note: candidate template ignored: couldn't infer template argument 'T'
  126 |     Ref(HashTableEmptyValueType) : m_ptr(hashTableEmptyValue()) { }
/Users/richardrobinson/Build/Products/Debug/usr/local/include/wtf/Ref.h:70:7: note: candidate template ignored: could not match 'Ref<T, type-parameter-0-1, type-parameter-0-2>' against 'MappedPeekType' (aka 'WebCore::Range *')
   70 | class Ref {
/Users/richardrobinson/Build/Products/Debug/usr/local/include/wtf/Ref.h:90:5: note: candidate template ignored: could not match 'Ref<T, type-parameter-0-1, type-parameter-0-2>' against 'MappedPeekType' (aka 'WebCore::Range *')
   90 |     Ref(const Ref& other)
/Users/richardrobinson/Build/Products/Debug/usr/local/include/wtf/Ref.h:95:38: note: candidate template ignored: could not match 'Ref<type-parameter-0-3, type-parameter-0-4>' against 'MappedPeekType' (aka 'WebCore::Range *')
   95 |     template<typename X, typename Y> Ref(const Ref<X, Y>& other)
]
```

getting these errors after doing so 😔

Reply:
`RefPtr`?